async-servlet-nio-application/build.gradle

```diff
@@ -46,3 +67,52 @@ publishing {
         }
     }
 }
+
+// Add a functional test task that runs during the 'check' phase that uses the Liberty Gradle plugin
+// to launch Liberty and run the test classes in the fvt source set
+task fvt(type: Test, dependsOn: jar) {
+    group 'Verification'
+    description 'Runs the functional verification tests.'
+    reports.html.destination = file("$buildDir/reports/fvt")
+    reports.junitXml.destination = file("$buildDir/test-results/fvt")
+    include '**/fvt/**'
+    exclude '**/unit/**'
+    systemProperties = ['liberty.test.port': getHttpPort()]
+    dependsOn publishWar, libertyStart
+    finalizedBy libertyStop
+}
+
+test {
+    reports.html.destination = file("$buildDir/reports/unit")
+    reports.junitXml.destination = file("$buildDir/test-results/unit")
+    include '**/unit/**'
+    exclude '**/fvt/**'
+}
+
+String getHttpPort() {
+    def DEFAULT_HTTP_PORT = 9083
+    def serverXmlFile = file(new File('../async-servlet-nio-wlpcfg/servers/servletNioSample/server.xml'))
+    def portPattern = ~/.*httpPort="(\d+)".*/
+    serverXmlFile.eachLine { line -> 
+        def portMatcher = line =~ portPattern
+        if (portMatcher) {
+            return portMatcher.group(1)
+        }
+    }
+    return DEFAULT_HTTP_PORT
+}
+
+libertyStart.mustRunAfter publishWar
+
+liberty {
+    wlpDir = owner.hasProperty('libertyRoot') ? "${libertyRoot}" : null
+    serverName = 'servletNioSample'
+    userDir = '../async-servlet-nio-wlpcfg'
+    clean = true
+}
+
+if (hasProperty('libertyRoot') && file("${libertyRoot}").exists() && file("${libertyRoot}/lib/ws-launch.jar").exists()) {
```

Comment:
I think this line will still fail if libertyRoot hasn't been set, that's why I did it with a default property and set it in ext, not 100% sure but would be worth checking.